httpx/_auth.py

```diff
@@ -141,6 +142,34 @@ def _build_auth_header(
         return f"Basic {token}"
 
 
+class NetRCAuth(Auth):
+    """
+    Use a 'netrc' file to lookup basic auth credentials based on the url host.
+    """
+
+    def __init__(self, file: typing.Optional[str]):
+        self._netrc_info = netrc.netrc(file)
+
+    def auth_flow(self, request: Request) -> typing.Generator[Request, Response, None]:
+        auth_info = self._netrc_info.authenticators(request.url.host)
+        if auth_info is None or auth_info[2] is None:
```

Comment:
I'm sorry, may I ask you why you are checking the password for `None`?

Reply:
I'm asking since previously you were checking 
```python
if credentials is not None:
    ...
```

and python documentation says ([link](https://docs.python.org/3/library/netrc.html#netrc.netrc.authenticators)) that 
> If the netrc file did not contain an entry for the given host, return the tuple associated with the ‘default’ entry. If neither matching host nor default entry is available, return None.

Reply:
Thanks @cdeler.

The simple version of "why" is because I was blindly following our previous implementation details here.

I've had a look into it, and I don't think the password can be `None` here, but I think it can be the empty string. It looks to me like Python 3.11+ will allow the empty string for a missing password field in the `netrc` file, and the previous versions will raise a parse error for missing passwords.

I think the robust thing to do here would be...

```python
if auth_info is None or not auth_info[2]:
```